pilot/pkg/xds/sds.go

```diff
@@ -138,7 +138,7 @@ func (s *SecretGen) Generate(proxy *model.Proxy, w *model.WatchedResource, req *
 	// SDS flow. The pilotSDSCertificateErrors metric and logs handle visibility into invalid references.
 	resources := filterAuthorizedResources(s.parseResources(w.ResourceNames, proxy), proxy, proxyClusterSecrets)
 
-	results := model.Resources{}
+	var results model.Resources
```

Comment:
It is very tricky to rely on nil check for a slice. Can we fix the caller by checking len(resources), I vagulaly remember different generator have different behavior for nil.

Reply:
Some generators intentionally return on 0 like CDS and LDS I think. But agree it should be more explicit (and investigate more which do and don't need it)

Reply:
Yes, i'd like to deep dive to see if we can align all that later